### PR TITLE
Refered by hedvig forever

### PR DESCRIPTION
**Pull request by FredrikAreschoug**
[Notion ticket](https://www.notion.so/hedviginsurance/71cf10addf9e42a2ae4e3e11d1d47a5d?v=698b1693c73d488996e38a2f36af5e58&p=583f4e1efcb545488c488fb614671457)

### CODE REVIEW

File: src/main/java/com/hedvig/productPricing/campaign/domain/model/RedeemedCampaign.kt [hunk 1]
```diff
@@ -51,29 +51,28 @@ open class RedeemedCampaign(
...
+        )) {
+          MemberStatus.SIGNED -> redeemedByStatus
+          MemberStatus.TERMINATED -> RedeemedCampaignStatus.TERMINATED
+          MemberStatus.NOT_SIGNED -> when (redeemedByStatus) {
+            RedeemedCampaignStatus.ACTIVE -> RedeemedCampaignStatus.OFFER
+            else -> redeemedByStatus
+          }
         }
       }
+      else -> redeemedByStatus
```
Comment (meletios) [inline src/main/java/com/hedvig/productPricing/campaign/domain/model/RedeemedCampaign.kt:75]: Nice!

File: src/main/java/com/hedvig/productPricing/campaign/service/EnableHedvigForeverService.kt [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+import com.hedvig.productPricing.campaign.domain.model.Campaign
+import com.hedvig.productPricing.campaign.domain.model.Member
 
 interface EnableHedvigForeverService {
   fun enableHedvigForever(
     market: Market,
     members: List<String>
   ): List<String>
+
+  fun getRedeemedHedvigForeverCampaignOrNull(member: Member): Campaign?
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/productPricing/campaign/service/EnableHedvigForeverService.kt:13]: Naming code after marketing is a no no IMO
```suggestion
  fun getRedeemedForeverCampaignOrNull(member: Member): Campaign?
```
```suggestion
  fun getRedeemedCustomReferralCampaignOrNull(member: Member): Campaign?
```
Or something of the like

Comment (FredrikAreschoug) [reply src/main/java/com/hedvig/productPricing/campaign/service/EnableHedvigForeverService.kt:13]: Then shouldn't we rename the service as well? But `CustomReferralCampaign` is not super clear for me what do you think about `EnableReferralsCampaign`

File: src/main/java/com/hedvig/productPricing/campaign/domain/model/RedeemedCampaign.kt (hunk 1 above)
Comment (johantjellden) [reply src/main/java/com/hedvig/productPricing/campaign/domain/model/RedeemedCampaign.kt:75]: 🥇

**Pull request merged**